Docs/reference/content/reference/driver/crud/sessions-and-transactions.md

```diff
@@ -0,0 +1,228 @@
+
+### IClientSession properties and methods
+
+The IClientSession interface defines the properties and methods available on a session.
+
+```csharp
+public interface IClientSession : IDisposable
+{
+    IMongoClient Client { get; }
+    BsonDocument ClusterTime { get; }
+    bool IsInTransaction { get; }
+    BsonTimestamp OperationTime { get; }
+    ClientSessionOptions Options { get; }
+
+    void AdvanceClusterTime(BsonDocument newClusterTime);
+    void AdvanceOperationTime(BsonTimestamp newOperationTime);
+
+    // see also transaction related methods documented below
+}
+```
+Note: a few members of IClientSession have been deliberately omitted from this documentation, either because they are rarely used or because they are for internal use only.
+
+#### Client
+
+The Client property returns a reference to the IMongoClient instance that was used to start this session.
+
+#### ClusterTime
+
+The ClusterTime property returns the highest cluster time that has been seen by this client. The value is an opaque BsonDocument containing a cluster time that has been returned by the server. While an application might never inspect the actual value, it might set this value (by calling AdvanceClusterTime) when initializing a causally consistent session.
+
+### IsInTransaction
+
+Specifies whether the session is currently in a transaction. A session is in a transaction after StartTransaction has been called and until either AbortTransaction or CommitTransaction has been called.
+
+### OperationTime
+
+The operation time returned by the server for the most recent operation in this session. Operation times are used by the driver to ensure causal consistency when the ClientSessionOptions specify that causal consistency is desired. While an application might never use the actual value, it might set this value (by calling AdvanceOperationTime) when initializing a causally consistent session.
+
+### Options
+
+Returns the options that were passed to StartSession.
+
+### AdvanceClusterTime and AdvanceOperationTime
+
+```csharp
+void AdvanceClusterTime(BsonDocument newClusterTime);
+void AdvanceOperationTime(BsonTimestamp newOperationTime);
+```
+
+Call these methods to advance the cluster and operation times when you want subsequent operations in this session to be causally consistent with operations that have executed outside of this session. Typically the values you pass to AdvanceClusterTime and AdvanceOperation time will come from the ClusterTime and OperationTime properties of some other session.
+
+### Transaction methods
+
+The methods to start, abort or commit a transaction are documented in the next section.
+
+## Transactions
+
+Transactions are started, committed or aborted using methods of IClientSession. A session can only be executing one transaction at a time, but a session can be used to execute multiple transactions as long as each transaction is committed or aborted before the next one is started.
```

Comment:
Minor quibble: perhaps "a session can execute one transaction at a time..."? Unsure if that accurately captures the meaning you were trying to convey though.

Reply:
Revised